mirgecom/boundary.py

```diff
@@ -62,13 +61,22 @@ def __init__(self, userfunc):
 
     def boundary_pair(self, discr, q, btag, **kwargs):
         """Get the interior and exterior solution on the boundary."""
+        ext_soln = self.exterior_soln(discr, q, btag, **kwargs)
+        int_soln = discr.project("vol", btag, q)
+        return TracePair(btag, interior=int_soln, exterior=ext_soln)
+
+    def exterior_soln(self, discr, q, btag, **kwargs):
+        """Get the exterior solution on the boundary."""
         actx = q[0].array_context
 
         boundary_discr = discr.discr_from_dd(btag)
         nodes = thaw(actx, boundary_discr.nodes())
         ext_soln = self._userfunc(nodes, **kwargs)
-        int_soln = discr.project("vol", btag, q)
-        return TracePair(btag, interior=int_soln, exterior=ext_soln)
+        return ext_soln
+
+    def av(self, discr, q, btag, **kwargs):
+        """Get the exterior solution on the boundary."""
+        return discr.project("vol", btag, q)
```

Comment:
I dig the changes that separate out the `exterior_soln` call, but I can't say the same for the AV-specific, and kind of vaguely-named `av` function. That an AV-specific function needs to go here at all seems (imo) to emit a code smell that is other than groovy.

Could we do this better with something like an "AV boundary wrapper" that inherits existing boundary objects and overloads, or replaces the boundary object methods it needs? e.g. `exterior_soln`.

Reply:
Yeah, so av is objectively a pretty terrible function name that I should have changed to something actually useful. 
![image](https://user-images.githubusercontent.com/26756513/111197060-01f9dd00-858c-11eb-9a8f-2ab8f100f296.png)
![image](https://user-images.githubusercontent.com/26756513/111197128-18a03400-858c-11eb-83ad-962d3826af17.png)
The `exterior_soln` call is then used to set the boundary conditions on Q in the calculation of R, and the terribly named `av` to set the conditions on R when calculating the RHS contribution that is made by the artificial viscosity. It's not clear to me if the conditions on R and Q are necessarily going to be the same. Particularly if more complex boundary conditions were to exist (characteristic, moving wall, ). It still might be possible to roll these into something cleaner though, I will think about it a bit and might pick your brain some later.